pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -128,19 +136,27 @@ def _matrix(cls, *params):
     def _eigvals(cls, *params):
         return cls.eigvals
 
-    def diagonalizing_gates(self):
-        r"""Rotates the specified wires such that they
-        are in the eigenbasis of the Pauli-X operator.
+    @staticmethod
+    def compute_diagonalizing_gates(wires):
+        r"""Diagonalizing gates of this operator.
 
-        For the Pauli-X operator,
+        These gates rotate the specified wires such that they
+        are in the eigenbasis of PauliX:
 
         .. math:: X = H^\dagger Z H.
```

Comment:
This is amazing 💯 (and actually a huge effort to do!)

Only two very minor thoughts:

- except for this docstring(?) we have placed other numerical representation in the _class_ docstring, since this is much more visible to readers, e.g., ![image](https://user-images.githubusercontent.com/2959003/146214181-06fe94a5-1801-4ee5-a385-9f7eaab9ce05.png), so this would mark a change in convention.

- it feels slightly odd to use inheritance here, but then redefine the docstring (since part of the nicety of inheritance is inheriting the docstring!). But this was the case _before_ your PR, your PR actually tidies it up and makes it a lot nicer.

Reply:
I personally think we should add examples and specific docstrings whereever possible, even if it multiplies the information? No harm in having the equation in the class docstring as well... and even after adding the examples now, I consulted them a lot to write tests! :)

Reply:
Yes you are right!